sopt/src/main/scala/dagr/sopt/cmdline/CommandLineParser.scala

```diff
@@ -0,0 +1,527 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015-2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package dagr.sopt.cmdline
+
+import dagr.commons.reflect.ReflectionUtil
+import dagr.sopt.clp
+import dagr.sopt.util.ParsingUtil._
+import dagr.commons.util.StringUtil._
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable.ListBuffer
+import scala.collection.{mutable, Set, Map}
+import scala.reflect.runtime.universe._
+import scala.reflect.ClassTag
+
+/** Various constants and methods for formatting and printing usages and error messages on the command line */
+trait CommandLineParserStrings {
+
+  /** Lengths for names and descriptions on the command line */
+  val ClpGroupNameColumnLength = 48
+  val ClpGroupDescriptionColumnLength = 45
+  val ClpNameColumnLength = 45
+
+  /** The maximum line lengths for tool descriptions */
+  val MaximumLineLength = 80
+
+  /** Error messages */
+  val AvailableClps = "Available Commands:"
```

Comment:
If we're going to let `sopt` depend on configuration, perhaps we should consider making all these strings configurable?  
![hrm](https://cloud.githubusercontent.com/assets/1609210/13050105/3a3eecf2-d3bf-11e5-8c5e-d9a4d5836f71.jpg)